pilot/pkg/bootstrap/sidecarinjector.go

```diff
@@ -15,39 +15,73 @@
 package bootstrap
 
 import (
+	"bytes"
 	"fmt"
+	"io/ioutil"
 	"os"
 	"path/filepath"
+	"time"
+
+	"github.com/hashicorp/go-multierror"
+	"k8s.io/api/admissionregistration/v1beta1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+
+	"istio.io/istio/pkg/util"
+	"istio.io/pkg/env"
 
 	"istio.io/istio/pkg/kube/inject"
 	"istio.io/pkg/log"
 )
 
+var (
+	webhookConfigName = env.RegisterStringVar("WEBHOOK", "",
+		"Name of webhook config to patch, if istioctl is not used.")
+)
+
+const (
+	// Name of the webhook config in the config - no need to change it.
+	webhookName = "sidecar-injector.istio.io"
+)
+
+// Was not used in Pilot for 1.3/1.4 (injector was standalone).
+// In 1.5 - used as part of istiod, if the inject template exists.
 func (s *Server) initSidecarInjector(args *PilotArgs) error {
+	// Injector should run along, even if not used - but only if the injection template is mounted.
+	// ./var/lib/istio/inject - enabled by mounting a template in the config.
 	injectPath := args.InjectionOptions.InjectionDirectory
 	if injectPath == "" {
-		log.Infof("Skipping sidecar injector, injection path not specified")
+		log.Infof("Skipping sidecar injector, injection path is missing")
 		return nil
 	}
+
 	// If the injection path exists, we will set up injection
-	if _, err := os.Stat(injectPath); !os.IsNotExist(err) {
-		dir, err := pilotDNSCertDir()
-		if err != nil {
-			return err
-		}
+	if _, err := os.Stat(filepath.Join(injectPath, "config")); !os.IsNotExist(err) {
 		parameters := inject.WebhookParameters{
-			ConfigFile: filepath.Join(injectPath, "config"),
-			ValuesFile: filepath.Join(injectPath, "values"),
-			MeshFile:   args.Mesh.ConfigFile,
-			CertFile:   filepath.Join(dir, "cert-chain.pem"),
-			KeyFile:    filepath.Join(dir, "key.pem"),
-			Port:       args.InjectionOptions.Port,
+			ConfigFile:          filepath.Join(injectPath, "config"),
+			ValuesFile:          filepath.Join(injectPath, "values"),
+			MeshFile:            args.Mesh.ConfigFile,
+			CertFile:            filepath.Join(DNSCertDir, "cert-chain.pem"),
+			KeyFile:             filepath.Join(DNSCertDir, "key.pem"),
+			Port:                args.InjectionOptions.Port,
+			HealthCheckFile:     "",
+			HealthCheckInterval: 0,
+			MonitoringPort:      s.basePort + 16, // TODO: disable the second monitoring port
 		}
 
 		wh, err := inject.NewWebhook(parameters)
 		if err != nil {
 			return fmt.Errorf("failed to create injection webhook: %v", err)
 		}
+		if webhookConfigName.Get() != "" {
+			s.addStartFunc(func(stop <-chan struct{}) error {
+				if err := patchCertLoop(s.kubeClient, stop); err != nil {
```

Comment:
@howardjohn I remember you removed it when introducing injector under pilot

Reply:
The long term plan is to have the cert managed by operator or istioctl, but for the short term we need to do this to get things moving. I had a similar PR just adding the cert patching, but I closed it since its conflicting with this change

Reply:
Yes, we may still iterate on this but feature parity first.